std/haxe/macro/ExprTools.hx

```diff
@@ -130,7 +130,8 @@ class ExprTools {
 					f(edef);
 			case EFormat(parts):
 				for (part in parts) switch part.kind {
-					case FRaw(_) | FIdent(_):
+					case FRaw(_):
+					case FIdent(i): f({pos: part.pos, expr: EConst(CIdent(i))});
```

Comment:
One thing that's bugging me a bit is that the fake `Ident` position here include the dollar sign, which is inconsistent with the expansion where we do the +1, but we can't really do the same in `ExprTools`, because we'd have to convert position with `getPosInfos`/`makePosition`, which seems too much. Or not?